pkg/controller/endpointslice/endpointslice_controller.go

```diff
@@ -46,6 +46,7 @@ import (
 	endpointutil "k8s.io/kubernetes/pkg/controller/util/endpoint"
 	endpointsliceutil "k8s.io/kubernetes/pkg/controller/util/endpointslice"
 	"k8s.io/kubernetes/pkg/features"
+	nodeutil "k8s.io/kubernetes/pkg/util/node"
```

Comment:
True, I was focusing on API surface for this change and this was the simplest way to not expose `NodeReady` from `topologycache`.

I think it's a reasonable standalone change, but I can also omit it since it'll have to addressed in a later PR. Don't have a strong preference.

Reply:
you can copy the function here and use it without exporting in  and avoid the dependency
```
// IsNodeReady returns true if a node is ready; false otherwise.
func IsNodeReady(node *v1.Node) bool {
	for _, c := range node.Status.Conditions {
		if c.Type == v1.NodeReady {
			return c.Status == v1.ConditionTrue
		}
	}
	return false
}
```

Reply:
Agree that we probably don't want to rely on nodeutil here

Reply:
Ack, removed the dependency and went with @aojea's suggestion.